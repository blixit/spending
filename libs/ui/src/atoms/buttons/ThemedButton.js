import React from 'react';
import styled from 'styled-components';

import ButtonComponent from './button/Button';

const Button = styled(ButtonComponent)`
  border: 0;
  border-radius: 3px;
  outline: none;
  &:hover{
    box-shadow: 1px 1px 3px 1px silver;
  }
  background-color: ${({ theme, color }) => color ? theme[color]() : '' }
  color: ${({ theme, color }) => color ? theme.text[color]() : '' }
`;

const ThemedButton = props => <Button {...props} />;

export default ThemedButton;
