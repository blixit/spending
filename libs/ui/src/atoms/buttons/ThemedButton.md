```js
  <ThemedButton>My ThemedButton</ThemedButton>
  <ThemedButton color='primary'>My ThemedButton</ThemedButton>
```