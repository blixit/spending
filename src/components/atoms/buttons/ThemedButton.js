import React from 'react';
import styled from 'styled-components';

import ButtonComponent from './Button';

const Button = styled(ButtonComponent)`
  border: 0;
  border-radius: 3px;
  outline: none;
  &:hover{
    box-shadow: 1px 1px 10px 1px silver;
  }
  background-color: ${({ theme, color }) => color ? theme[color]() : '' }
  color: ${({ theme, color }) => color ? theme.text[color]() : '' }
`;

const ThemedButton = (props) => {

  return (
    <Button {...props} />
  );
}

export default ThemedButton;
