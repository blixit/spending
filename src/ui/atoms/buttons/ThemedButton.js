import React from 'react';
import styled from 'styled-components';

import { guessTextColor } from 'ui/theming/theme';

import ButtonComponent from './button/Button';

const Button = styled(ButtonComponent)`
  border: 0;
  border-radius: 3px;
  outline: none;
  &:hover{
    box-shadow: 1px 1px 3px 1px silver;
  }
  background-color: ${({ theme, bgcolor }) => {
    if (!bgcolor) { return ''; }
    return theme[bgcolor] ? theme[bgcolor]() : bgcolor;
  }};
  color: ${props => guessTextColor(props)};
`;

const ThemedButton = props => <Button {...props} />;

ThemedButton.displayName = 'ThemedButton';

export default ThemedButton;
