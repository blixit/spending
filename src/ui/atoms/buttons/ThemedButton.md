```js
<h1>Setting only bgcolor</h1>
<>
  <ThemedButton>My ThemedButton</ThemedButton>
  <pre />
  <ThemedButton bgcolor='primary'>My ThemedButton</ThemedButton>
  <pre />
  <ThemedButton bgcolor='secondary'>My ThemedButton</ThemedButton>
  <pre />
  <ThemedButton bgcolor='third'>My ThemedButton</ThemedButton>
  <pre />
  <ThemedButton bgcolor='fourth'>My ThemedButton</ThemedButton>
  <pre />
  <ThemedButton bgcolor='fifth'>My ThemedButton</ThemedButton>
</>
<pre />
<h1>Setting color and bgcolor (specifig theming)</h1>
<>
  <ThemedButton color='brown' bgcolor='lightBlue' >My ThemedButton</ThemedButton>
</>
```