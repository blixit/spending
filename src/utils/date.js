/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};