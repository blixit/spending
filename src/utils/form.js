/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
