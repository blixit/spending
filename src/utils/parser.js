export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);