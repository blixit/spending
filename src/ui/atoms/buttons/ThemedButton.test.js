import 'jest-styled-components'

import render from 'testing/render';

import { withTheme } from 'ui/theming/test.utils';
import ThemedButton from './ThemedButton';

describe('[atoms/buttons] ThemedButton', () => {
  const props = {
    bgcolor: 'yellow',
    children: 'button text'
  };

  it('should render properly', () => {
    const tree = render(withTheme(ThemedButton, props));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
