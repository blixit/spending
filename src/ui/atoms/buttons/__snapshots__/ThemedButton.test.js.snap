// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[atoms/buttons] ThemedButton should render properly 1`] = `
.c0 {
  min-height: 24px;
  lineheight: 25px;
  padding: 5px;
  cursor: pointer;
}

.c1 {
  border: 0;
  border-radius: 3px;
  outline: none;
  background-color: yellow;
  color: yellow;
}

.c1:hover {
  box-shadow: 1px 1px 3px 1px silver;
}

<button
  className="c0 c1"
>
  button text
</button>
`;
